nmrfx-core/src/main/java/org/nmrfx/datasets/DatasetRegion.java

```diff
@@ -281,6 +281,10 @@ public int compare(Object o1, Object o2) {
                 result = -1;
             } else if (r2 == null) {
                 result = 1;
+            } else if (r1.x == null) {
+                result = -1;
+            } else if (r2.x == null) {
+                result = 1;
```

Comment:
added these two conditions to avoid NPE. This change means that if two DatasetRegions have the parameter x as null, they will not be considered equal. Should they be considered equal? (before there would have been an exception thrown during the comparison). I didn't want to make too many changes as there is this note on the file indicating that the equals method may not be correct anyways: https://github.com/nanalysis/nmrfx/blob/4145ec117ba89f6926ff2be1dc000cdcaaa25900/nmrfx-core/src/main/java/org/nmrfx/datasets/DatasetRegion.java#L9